### PR TITLE
Added support for privilege key.

### PR DESCRIPTION
**Pull request by krishna-patel-ch**
__Short description outlining key changes/additions__

Added support for privilege key.

__JIRA Ticket Number__

IDVA6-673

### Type of change

* [ ] Bug fix
* [x] New feature
* [ ] Breaking change
* [x] Infrastructure change

### Pull request checklist

* [x] I have added unit tests for new code that I have added
* [x] I have added/updated functional tests where appropriate
* [x] The code follows our [coding standards](https://github.com/companieshouse/styleguides/blob/master/java.md)

__An exhaustive list of peer review checks can be read [here](https://github.com/companieshouse/styleguides/blob/master/java_review.md#developer-actions-prior-to-code-commitreview-started)__

### CODE REVIEW

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

File: src/main/java/uk/gov/companieshouse/accounts/user/configuration/InterceptorConfig.java [hunk 1]
```diff
@@ -51,6 +52,9 @@ private void addEricInterceptors( final InterceptorRegistry registry){
                                 AccountsUserServiceApplication.applicationNameSpace)
                 )
         ).excludePathPatterns("/*/healthcheck");
```
Comment (kkonuganti-ch) [inline src/main/java/uk/gov/companieshouse/accounts/user/configuration/InterceptorConfig.java:54]: this is good, but do you think we should disable the eric interceptor to not allow any user to access with internal api key privilege

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Review by kkonuganti-ch — APPROVED** [review]

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Pull request merged**